config/validation_test.go

```diff
@@ -0,0 +1,298 @@
+package config
+
+import (
+	. "gopkg.in/check.v1"
+)
+
+var (
+	defaultBackends = &BackendDrivers{CRUD: "ceph", Mount: "ceph", Snapshot: "ceph"}
+	VolumeConfigs   = map[string]map[string]*Volume{
+		"valid": {
+			"basic": {
+				DriverOptions:  map[string]string{"pool": "rbd"},
+				CreateOptions:  CreateOptions{Size: "10MB"},
+				RuntimeOptions: RuntimeOptions{UseSnapshots: false},
+				VolumeName:     "basicvolume",
+				PolicyName:     "basicpolicy",
+				Backends:       defaultBackends,
+			},
+			"basicwithruntime": {
+				DriverOptions:  map[string]string{"pool": "rbd"},
+				CreateOptions:  CreateOptions{Size: "10MB"},
+				RuntimeOptions: RuntimeOptions{UseSnapshots: true, Snapshot: SnapshotConfig{Frequency: "10m", Keep: 10}},
+				VolumeName:     "basicvolume",
+				PolicyName:     "basicpolicy",
+				Backends:       defaultBackends,
+			},
+		},
+		"invalid": {
+			"emptyvolume": {}, // base case
+			"emptyname": {
+				VolumeName:     "",
+				PolicyName:     "policy",
```

Comment:
add a test case in this area for names with a / in them (policy or volume). that is also invalid.